### PR TITLE
Sortable table v3 task solution

### PR DESCRIPTION
**Pull request by baidauletov**
### Sortable table 
- Materials from taksPicker about infinite scroll, especially about window getBoundingClientRect() was useful for this task
- Update function idea from Column chart task used to update rows in table
- Loading property used to stop fetch new data until previous request resolving

### CODE REVIEW

**Comment by jsru** [comment]
Добавляю преподавателя (@dosandk) для код-ревью.

File: 06-events-practice/2-tooltip/index.js [hunk 1]
```diff
@@ -23,19 +23,30 @@ class Tooltip {
 
     if (node) {
       this.render(node.dataset.tooltip);
+      document.addEventListener("pointermove", this.pointerMove);
     }
   };
```
Comment (dosandk) [inline 06-events-practice/2-tooltip/index.js:29]: good 👍

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 2]
```diff
@@ -1,20 +1,315 @@
...
+  loading = false;
 
+  constructor(
+    headersConfig = [],
+    {
+      url = "",
+      isSortLocally,
+      data = [],
+      sorted = {
+        id: headersConfig.find((item) => item.sortable).id,
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:20]: good 👍

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 3]
```diff
@@ -1,20 +1,315 @@
...
+      this.end
+    );
+
+    this.updateRows(data);
+
+    this.initEventListeners();
+  }
+
+  getSubElements(element) {
+    let result = {};
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:62]: `const result`

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 4]
```diff
@@ -1,20 +1,315 @@
...
+    loadingNode.dataset.element = "loading";
+
+    return loadingNode;
+  }
+
+  async fetchData(url) {
+    return await fetchJson(url);
+  }
+
+  windowScroll = async () => {
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:164]: good 👍

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 5]
```diff
@@ -1,20 +1,315 @@
...
+    const { id, order } = this.sorted;
+    const clientHeight = document.documentElement.clientHeight;
+    let windowRelativeBottom =
+      document.documentElement.getBoundingClientRect().bottom;
+
+    if (windowRelativeBottom < clientHeight + 100 && !this.loading) {
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:170]: 100 было бы неплохо вынести в переменную с говорящим именем

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 6]
```diff
@@ -1,20 +1,315 @@
...
+
+      const data = await this.loadData(id, order, this.start, this.end);
+
+      this.updateRows(data);
+
+      this.loading = false;
+    }
+  };
+
+  async loadData(id, order, start, end) {
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:181]: 👍 good

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 7]
```diff
@@ -1,20 +1,315 @@
...
+    this.subElements = this.getSubElements(this.element);
+
+    this.element.classList.add("loading-line", "sortable-table__loading-line");
+
+    const shift = 30;
+    this.start = this.end;
+    this.end += shift;
+
+    let url = this.url;
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:190]: `const url`

File: 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js [hunk 8]
```diff
@@ -1,20 +1,315 @@
...
+
+    url.searchParams.set("_sort", id);
+    url.searchParams.set("_order", order);
+    url.searchParams.set("_start", start);
+    url.searchParams.set("_end", end);
+
+    const response = await this.fetchData(url);
```
Comment (dosandk) [inline 07-async-code-fetch-api-part-1/2-sortable-table-v3/index.js:197]: 👍 good

**Review by dosandk — APPROVED** [review]
👍 good

**Pull request merged**